### PR TITLE
Setting up template to use 1ES hosted pool.

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR modifies the Python pipelines to use the 1ES hosted pools for WIndows/Linux build/test legs. It has a few aspects:

1. Change to the stage template to define a WindowsPool and LinuxPool.
2. Change to the job template to define a WindowsPool and LinuxPool
3. Maps WindowsLinux and LinuxPool to variables in the test job so that they can be used in the matrix.

### CODE REVIEW

File: eng/pipelines/templates/steps/build-test.yml [hunk 1]
```diff
@@ -24,8 +24,6 @@ steps:
     parameters:
       versionSpec: '${{ parameters.PythonVersion }}'
 
-  - template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/build-test.yml:27]: I'm OK with us disabling this temporarily to onboard the pool but please file an issue to track enabling some way to validate the machines are as we expect. We should either have something that runs in the pipeline or have some external pipeline that can run on the different pools to assert they are giving use the correct OS images.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/steps/build-test.yml (hunk 1 above)
Comment (mitchdenny) [reply eng/pipelines/templates/steps/build-test.yml:27]: https://github.com/Azure/azure-sdk-tools/issues/1313

**Pull request merged**